--- setup.py
+++ setup.py
@@ -12,7 +12,8 @@
 	author_email='smccarty@eyemg.com',
 	url='http://opensource.eyemg.com/index.php/Petit',
 	scripts=["bin/petit"], 
-	py_modules=['lib.crunchtools'],
+	packages=['crunchtools'],
+	package_dir={'': 'lib'},
 	data_files=[	('/var/lib/petit/fingerprints',glob.glob(os.path.join('lib','fingerprints','*.fp'))),
 			('petit/lib/fingerprint_library', glob.glob(os.path.join('lib','fingerprint_library','*.fp'))),
 			('/var/lib/petit/filters', glob.glob(os.path.join('lib','filters','*.stopwords')))]
