diff --git a/runtime/staprun/staprun_funcs.c b/runtime/staprun/staprun_funcs.c
index 4e525b0..0398b5e 100644
--- a/runtime/staprun/staprun_funcs.c
+++ b/runtime/staprun/staprun_funcs.c
@@ -448,7 +448,7 @@ check_groups (void)
  *
  * Returns: -1 on errors, 0 on failure, 1 on success.
  */
-int check_permissions(const void *module_data, off_t module_size)
+int check_permissions(__attribute__((unused)) const void *module_data, __attribute__((unused)) off_t module_size)
 {
 	int check_groups_rc;
 	int check_signature_rc = 0;
