diff -Nur terminator-1.91.orig/terminatorlib/terminator.py terminator-1.91/terminatorlib/terminator.py
--- terminator-1.91.orig/terminatorlib/terminator.py	2017-02-28 22:07:22.000000000 +0000
+++ terminator-1.91/terminatorlib/terminator.py	2017-12-28 15:33:04.584351677 +0000
@@ -95,8 +95,6 @@
             self.doing_layout = False
         if not self.pid_cwd:
             self.pid_cwd = get_pid_cwd()
-        if self.gnome_client is None:
-            self.attempt_gnome_client()
         self.connect_signals()
 
     def connect_signals(self):
