diff -Nur terminator-1.91.orig/data/terminator.desktop.in terminator-1.91/data/terminator.desktop.in
--- terminator-1.91.orig/data/terminator.desktop.in	2017-02-26 17:45:29.000000000 +0000
+++ terminator-1.91/data/terminator.desktop.in	2017-12-28 15:35:23.894561899 +0000
@@ -10,7 +10,7 @@
 X-Ubuntu-Gettext-Domain=terminator
 X-Ayatana-Desktop-Shortcuts=NewWindow;
 Keywords=terminal;shell;prompt;command;commandline;
-[NewWindow Shortcut Group]
+[X-NewWindow Shortcut Group]
 Name=Open a New Window
 Exec=terminator
 TargetEnvironment=Unity
